Compiler erorr - expected usize but got u64

Compiler error in `rust-alloc/src/thread_cache.rs:46` - `libc::pthread_self()` expected usize but got u64

```
error[E0308]: mismatched types
  --> rust-alloc/src/thread_cache.rs:46:14
   |
45 | pub(crate) fn thread_id() -> usize {
   |                              ----- expected `usize` because of return type
46 |     unsafe { libc::pthread_self() }
   |              ^^^^^^^^^^^^^^^^^^^^ expected `usize`, found `u64`
   |
```

--- a/rust-alloc/src/thread_cache.rs
+++ b/rust-alloc/src/thread_cache.rs
@@ -43,7 +43,7 @@ fn hash_usize(input: usize) -> usize {
 
 #[cfg(unix)]
 pub(crate) fn thread_id() -> usize {
-    unsafe { libc::pthread_self() }
+    unsafe { libc::pthread_self().try_into().unwrap() }
 }
 
 #[cfg(windows)]

